api/mfx_dispatch/linux/mfxloader.cpp

```diff
@@ -177,10 +180,19 @@ mfxStatus LoaderCtx::Init(mfxInitParam& par)
     return MFX_ERR_UNSUPPORTED;
   }
 
+  eMFXHWType platform = MFX_HW_UNKNOWN;
+  auto devices = get_devices();
+  if (devices.size()) {
+    platform = devices[0].platform;
+  }
+
   std::vector<std::string> libs;
 
-  if (MFX_IMPL_BASETYPE(par.Implementation) == MFX_IMPL_AUTO ||
-      MFX_IMPL_BASETYPE(par.Implementation) == MFX_IMPL_AUTO_ANY) {
+  if (platform >= MFX_HW_ADL_P) {
```

Comment:
We need to avoid adding new platform IDs to the list. I.e. check needs to be: if platform is unknown - load VPL.